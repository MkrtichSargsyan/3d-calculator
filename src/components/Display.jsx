import styled from 'styled-components';
import PropTypes from 'prop-types';

const DisplayArea = styled.div`
  width: 400px;
  padding:25px 20px;
  background: grey;
  border-radius: 3px;
  color: white;
  text-align:right;
  margin: 0 auto;
`;

const Display = ({ result }) => <DisplayArea>{result}</DisplayArea>;

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
