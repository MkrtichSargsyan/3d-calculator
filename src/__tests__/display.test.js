import renderer from 'react-test-renderer';
import Display from '../components/Display';

it('expects to render display component', () => {
  const result = renderer.create(<Display />).toJSON;
  expect(result).toMatchSnapshot();
});
