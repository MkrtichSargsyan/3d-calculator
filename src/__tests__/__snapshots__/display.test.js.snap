// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render display component 1`] = `[Function]`;
